### PR TITLE
MSYS-827: Added test coverage

### PR DESCRIPTION
**Pull request by Nimesh-Msys**
- initial commit to tests 'Once' type schedulers functionality

Signed-off-by: Nimesh Patni <nimesh.patni@msystechnologies.com>

### CODE REVIEW

File: spec/win32/taskschedular_spec.rb [hunk 1]
```diff
@@ -0,0 +1,785 @@
...
+
+
+RSpec.describe Win32::TaskScheduler, :windows_only  do
+  let(:application) { "notepad.exe" }
+  let(:tsk_time) { Time.now }
+  let(:dummy_task_name) { "test_task_dummy" }
+  let(:trigger) { { start_year: tsk_time.year, start_month: tsk_time.month, start_day: tsk_time.day,
+                    start_hour: tsk_time.hour, start_minute: tsk_time.min } }
+
+  describe 'Ensuring trigger constants' do
```
Comment (Vasu1105) [inline spec/win32/taskschedular_spec.rb]: we can move this in unit test

File: spec/win32/taskschedular_spec.rb [hunk 2]
```diff
@@ -0,0 +1,785 @@
...
+    end
+  end
+
+  describe '#constructor' do
+    let(:ts) { Win32::TaskScheduler }
+    let(:task) { "test_task" }
+    let(:folder) { '\\' }
+    let(:force) { false }
+    before { trigger[:trigger_type] = Win32::TaskScheduler::ONCE }
+    context 'argument' do
```
Comment (Vasu1105) [inline spec/win32/taskschedular_spec.rb]: Move this in unit test

File: spec/win32/taskschedular_spec.rb [hunk 3]
```diff
@@ -0,0 +1,785 @@
...
+          expect{ ts.new(task, trigger, folder, force, 1) }.to raise_error(ArgumentError)
+          expect{ ts.new(task, trigger, folder, force, 'abc') }.to raise_error(ArgumentError)
+        end
+      end
+      context 'task' do
+        it 'default value is nil' do
+          task_scheduler = ts.new(task)
+          expect(task_scheduler).to be_a(ts)
+        end
+        it 'does not create any task if trigger is absent' do
```
Comment (Vasu1105) [inline spec/win32/taskschedular_spec.rb]: please have blank space between each test cases

File: spec/win32/taskschedular_spec.rb (hunk 1 above)
Comment (Nimesh-Msys) [reply spec/win32/taskschedular_spec.rb]: Done

File: spec/win32/taskschedular_spec.rb (hunk 2 above)
Comment (Nimesh-Msys) [reply spec/win32/taskschedular_spec.rb]: Done

File: spec/win32/taskschedular_spec.rb (hunk 3 above)
Comment (Nimesh-Msys) [reply spec/win32/taskschedular_spec.rb]: Done

File: spec/win32/taskschedular_spec.rb [hunk 4]
```diff
@@ -0,0 +1,588 @@
...
+      end
+      context 'folder' do
+        let(:service) { WIN32OLE.new('Schedule.Service') }
+        let(:root_path) { "\\" }
+        let(:test_path) { "\\Foo" }
+        before { service.Connect }
+        it 'default value is root' do
+          task_scheduler = ts.new(task, trigger)
+          expect(service.GetFolder(root_path).GetTask(task)).to be_a(WIN32OLE)
+          delete_task
```
[the remark this answers is not in the document]
Comment (Nimesh-Msys) [reply spec/win32/taskschedular_spec.rb]: Done

**Comment by Nimesh-Msys** [comment]
Have submitted #58 to cover this functionality.
This PR is not required anymore.
Closing.

**Pull request closed without merging**